addons/delivery/models/stock_move.py

```diff
@@ -19,8 +19,9 @@ def _cal_move_weight(self):
 
     def _get_new_picking_values(self):
         vals = super(StockMove, self)._get_new_picking_values()
-        carrier_id = self.group_id.sale_id.carrier_id.id
-        vals['carrier_id'] = any(propagate_carrier for propagate_carrier in self.rule_id) and carrier_id
+        if self.picking_type_id.code == 'outgoing':
+            carrier_id = self.group_id.sale_id.carrier_id.id
+            vals['carrier_id'] = any(propagate_carrier for propagate_carrier in self.rule_id) and carrier_id
```

Comment:
I don't understand why you're adding this in. If it's to prevent creating a shipping label for returns generated by a SO, this is already prevented in https://github.com/odoo/odoo/blob/d49246d709759f365355dadd022c9d0811d08b9a/addons/delivery/models/stock_picking.py#L178-L179

Reply:
Hello @ticodoo  while we create a sale order line with minus qty it will create a return transfer with a set carrier. so we only set the shipping carrier in only for "Delivery orders".

But don't worry I will remove it.

if I'll remove this code then the except same code is available in v14 PR, can you please check that PR?
https://github.com/odoo/odoo/pull/104658/files

Reply:
There are cases when we may want a return transfer to have a carrier (e.g. to create a return label, this may not 100% be working correctly right now though.. I need to double check) also this change would have conflicted with https://github.com/odoo/odoo/pull/62851 for future reference (i.e. it's easy to forget about the multi-step cases :crying_cat_face: )